modules/http4s/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/Http4sRegistryLookup.scala

```diff
@@ -94,5 +97,40 @@ object Http4sRegistryLookup {
       .fromString(s"${cxn.uri.toString.stripSuffix("/")}/schemas/$vendor/$name/jsonschema/$model")
       .leftMap(e => RegistryError.ClientFailure(e.message))
 
+  def runRequest[F[_]: Sync, A: EntityDecoder[F, *]](
+    client: HttpClient[F],
+    req: Request[F]
+  ): F[Either[RegistryError, A]] = {
+    val responseResult = client.run(req).use[F, Either[RegistryError, A]] {
+      case Status.Successful(response) =>
+        response.as[A].map(_.asRight)
```

Comment:
I think `.as[A]` can raise an error in `F` if the body cannot be parsed as expected.  (I'm only 80% certain on that - need to check).  Is it OK to let the error bubble up, or should this be caught immediately and converted to one of the failure classes?

Reply:
That's a good point. I think likely we are stumbling upon this infamous #152 because the body is trimmed.